.github/unittest.sh

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+# Prepare conda
+CONDA_PATH=$(which conda)
+eval "$(${CONDA_PATH} shell.bash hook)"
```

Comment:
IIUC, all the generic jobs come with `conda` pre-installed and use `bash`, right? If so, could we maybe run `conda init` before we enter the script? Otherwise stuff like `conda activate` doesn't work without running the highlighted commands before.

Reply:
At least the `conda` part is not true for Windows. @osalpekar is this intentional?

Reply:
It is not intentional - since this is the first application of the windows job, we'll need to identify and patch these gaps as we go :) This line would need to be added to the windows_job to ensure conda is installed: https://github.com/pytorch/test-infra/blob/main/.github/workflows/macos_job.yml#L84